src/cards.ts

```diff
@@ -6,9 +6,11 @@ export default class Cards {
     private imagesCount = imageNames.length;
     public cardsList: Card[] = [];
     public cardCount: number;
+    public sameCards: number;
 
-    constructor(cardCount: number) {
+    constructor(cardCount: number, sameCards: number) {
         this.cardCount = cardCount;
+        this.sameCards = sameCards;
     }
 
     public createCards() {
```

Comment:
Nee jestem przekonany by to było w jednym pliku. (u mnie w pracy tak robimy czasami ale mi się średnio to widzi).
Nie wiem czy nie lepiej by byo utworzyć z tego co poniżej jest klase service która by obsługiwała działania na obiekcie Cards a tutaj zostawić tylko samo Cards. Plus jak by jeszce interfejs do tego wszedł i service obsługiwał interfejs a nie sam obiekt cards to by była baja jak dla mnie bo wtedy kod fajnie rozszerzalny by był.
Pozostawiam to do twojej oceny bo w sumie do końca nie wiem jak się w JS pisze a to co tutaj sprzedaje to podejście typowo z C#